server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -1086,4 +1090,16 @@ private void oversizedCopy(BytesRef from, BytesRef to) {
      * Predicate used for {@link #acceptedGlobalOrdinals} if there is no filter.
      */
     private static final LongPredicate ALWAYS_TRUE = l -> true;
+
+    /**
+     * If DocValues have not been initialized yet for reduce phase, create and set them.
+     */
+    private SortedSetDocValues getDocValues() throws IOException {
+        if (dvs == null) {
```

Comment:
The reduce phase should be called only once, right? If yes, we could have `private final SetOnce<SortedSetDocValues> dvs;` ?

Reply:
Good idea, made the change in the latest revision!